src/document.c

```diff
@@ -3813,3 +3813,14 @@ void document_grab_focus(GeanyDocument *doc)
 
 	gtk_widget_grab_focus(GTK_WIDGET(doc->editor->sci));
 }
+
+static void        *copy_(void *src) { return src; }
+static void         free_(void *doc) { }
+
+/** Gets the GType of GeanyDocument
+ *
+ * @return the GeanyDocument type */
```

Comment:
hum… shouldn't those be `@gironly` too just yet?
they aren't really useful types as they don't to any instance management, so they are pointless in C (on advantage to use their type over `G_TYPE_POINTER`), so what about keeping that off API for the moment?

That would also limit the exposed surface so possibly limiting some future compatibility breakage.